DigitalLearningSolutions.Data/Services/PasswordResetService.cs

```diff
@@ -0,0 +1,121 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Data;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Exceptions;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Models.User;
+    using Microsoft.Extensions.Logging;
+    using MimeKit;
+
+    public interface IPasswordResetService
+    {
+        void SendResetPasswordEmail(string emailAddress);
+    }
+
+    public class PasswordResetService : IPasswordResetService
+    {
+        private const string AdminTableName = "AdminUsers";
+        private const string DelegatesTableName = "Candidates";
+        private const string AdminIdColumnName = "AdminID";
+        private const string DelegatesIdColumnName = "CandidateID";
+
+        private readonly IUserService userService;
+        private readonly IDbConnection connection;
+        private readonly ILogger<PasswordResetService> logger;
+        private readonly IConfigService configService;
+        private readonly IEmailService emailService;
+
+        public PasswordResetService(
+            IUserService userService,
+            IDbConnection connection,
+            ILogger<PasswordResetService> logger,
+            IConfigService configService,
+            IEmailService emailService)
+        {
+            this.userService = userService;
+            this.connection = connection;
+            this.logger = logger;
+            this.configService = configService;
+            this.emailService = emailService;
+        }
+
+        public void SendResetPasswordEmail(string emailAddress)
```

Comment:
I feel like this method is actually doing more than just sending an email. Maybe name is ResetUsersPassword or ResetPassword?

Reply:
I feel like "ResetUsersPassword" and "ResetPassword" might be a misnomer as it does not actually do any resetting of passwords, just sending an email about it. Would something like "GenerateResetPasswordLink" be more suitable?

Reply:
That's a fair point! I feel like GenerateResetPasswordLink would describe the first half of what the service does though. Maybe "HandlePasswordReset" (fairly generic) or "GenerateAndSendPasswordResetLink" (slightly clunky but fairly accurate)?